ui/src/ensembler/components/modal/DeleteEnsemblerModal.js

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef, useState } from "react";
+import { addToast } from "@caraml-dev/ui-lib";
+import { useEnsemblerModal } from "./useEnsemblerModal";
+import { useTuringApi } from "../../../hooks/useTuringApi";
+import { ConfirmationModal } from "../../../components/confirmation_modal/ConfirmationModal";
+import { isEmpty } from "../../../utils/object";
+import { ListEnsemblingJobsForEnsemblerTable } from "../table/ListEnsemblingJobsForEnsemblerTable";
+import { ListRouterVersionsForEnsemblerTable } from "../table/ListRouterVersionsForEnsemblerTable";
+import { EuiFieldText } from "@elastic/eui";
+
+export const DeleteEnsemblerModal = ({
+  onSuccess,
+  deleteEnsemblerRef,
+}) => {
+  const closeModalRef = useRef();
+
+  const [canDeleteEnsembler, setCanDeleteEnsembler] = useState(true)
+  const [ensemblerUsedByCurrentRouterVersion, setEnsemblerUsedByCurrentRouterVersion] = useState(false)
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+  const [ensembler = {}, openModal, closeModal] = useEnsemblerModal(closeModalRef);
+
+  const [{ isLoading, isLoaded, error }, submitForm] =  useTuringApi(
+    `/projects/${ensembler.project_id}/ensemblers/${ensembler.id}`,
+    {
+      method: "DELETE",
+      headers: { "Content-Type": "application/json" },
+    },
+    {},
+    false
+  );
+
+  useEffect(() => {
+    if (!isEmpty(ensembler)  && isLoaded && !error) {
+      addToast({
+        id: `submit-success-delete-${ensembler.id}`,
+        title: `Ensembler ${ensembler.name} has been deleted!`,
+        color: "success",
+        iconType: "check",
+      });
+      onSuccess();
+      closeModal();
+    }
+  }, [isLoaded, error, ensembler, onSuccess, closeModal]);
+
+  const updateStatus = (newStatus) => {
+    if ((canDeleteEnsembler && newStatus) || (!canDeleteEnsembler && !newStatus)) {
```

Comment:
* Per your logic, `canDeleteEnsembler` can be set to `false` even when `newStatus ` is `true`.  Is this intentional?
* For consistency, can this function be named to match the `canDeleteEnsembler` state? Also, I think the check can be more concise:
```javascript
const updateCanDeleteEnsembler = (newCanDeleteEnsembler) => {
  if (canDeleteEnsembler !== newCanDeleteEnsembler) {
    setCanDeleteEnsembler(false);
  }
}
```

Now, I have a more basic question - why do we even need this function? Can we directly pass the `setCanDeleteEnsembler` down? IIRC the change will only be triggered by `useState` if old and new values are not the same.

Reply:
the api call was inside a component and the main reason for the updateStatus function is to see if one of the api call (either ensembling job or router version, if one of them have condition that make the deletion of ensembler prohibited, then the other one won't show it's entity. for example : the ensembler have 2 active router version and 3 inactive ensembling jobs, since the router version block the deletion, the popup won't show the 3 inactive ensembling jobs.

But i understand of some confusion that i create, i have refactor the updateStatus to become 2 state, for each the router version and the ensembling jobs, please take a look. Thanks!